src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -385,112 +403,47 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema, go
-    def validateUpsertPermissions()(op: Boolean => PerRequestMessage) = {
-      val pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]] = authHeaders(requestContext) ~> sendReceive
-
-      val probeUpsert = pipeline {
-        Post(FireCloudDirectiveUtils.encodeUri(Rawls.entityPathFromWorkspace(workspaceNamespace, workspaceName) + "/batchUpsert"), List.empty[String])
-      }
-
-      probeUpsert.map {
-        case resp if resp.status == NoContent => op(true)
-        case resp => RequestComplete(resp) // bubble up errors
-      }
-    }
-
     // validate presigned URL
     validatePfbUrl(pfbRequest) { pfbUrl =>
-      // empty-array batchUpsert to Rawls to verify workspace existence and permissions
-      validateUpsertPermissions() { _ =>
-        // generate job UUID
-        val jobId = java.util.UUID.randomUUID()
-
-        // the payload to Arrow needs to include the PFB url, job id, workspace info, and user info
-        val user = WorkbenchUserInfo(userInfo.id, userInfo.userEmail)
-        val arrowPayload = pfbRequest.copy(jobId = Option(jobId.toString),
-          workspace = Option(WorkspaceName(workspaceNamespace, workspaceName)),
-          user = Option(user))
-
-        // fire-and-forget call Arrow with UUID, workspace info, and presigned URL
-        val idToken = googleServicesDAO.getAdminIdentityToken
-        val gzipPipeline = addHeader (`Accept-Encoding`(gzip)) ~> addCredentials(OAuth2BearerToken(idToken)) ~> sendReceive ~> decode(Gzip)
-        gzipPipeline { Post(avroToRawlsURL, arrowPayload) }
-
-        // the response payload to the end user omits the userid/email
-        val responsePayload = arrowPayload.copy(user = None)
-
-        // return Accepted with UUID
-        RequestComplete(Accepted, responsePayload)
+      // verify workspace existence and permissions
+      validateUpsertPermissions(userInfo, workspaceNamespace, workspaceName, "write") flatMap { _ =>
+        // the payload to Import Service sends the path
+        val importServicePayload: ImportServiceRequest = ImportServiceRequest(path = pfbUrl.toString, filetype = "pfb")
+        // TODO: AS-155: always encode namespace/name
+        val importServiceUrl = s"${FireCloudConfig.ImportService.server}/$workspaceNamespace/$workspaceName/imports"
+
+        userAuthedRequest(Post(importServiceUrl, importServicePayload))(userInfo) map {
+          case resp if resp.status == Created =>
+            val importServiceResponse = unmarshal[ImportServiceResponse].apply(resp)
+            // for backwards compatibility, we return Accepted(202), even though import service returns Created(201)
+            RequestComplete(Accepted, importServiceResponse)
+          case otherResp =>
+            RequestCompleteWithErrorReport(otherResp.status, otherResp.toString)
+        }
       }
     }
   }
 
   def pfbImportStatus(workspaceNamespace: String, workspaceName: String, jobId: String, userInfo: UserInfo): Future[PerRequestMessage] = {
```

Comment:
Currently, in order to get the status of a job, we need the jobId. It might be cool to be able to just ask what the statuses of all jobs in my workspace are, and only have to pass in the namespace and name of the workspace. This is not for this PR, of course, just something we could consider adding in the future.

Reply:
yes absolutely! That exists in import service, and is in my list of TODOs for "new endpoint for listing import jobs in a workspace".